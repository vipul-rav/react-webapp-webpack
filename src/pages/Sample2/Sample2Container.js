import React, { memo } from 'react';

const Sample2Container = memo(() => (
    <div>
        Sample 2 Component.<button>asasd</button>
    </div>
));

export { Sample2Container };
