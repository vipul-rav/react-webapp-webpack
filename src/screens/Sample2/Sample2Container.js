import React, { memo } from 'react';

const Sample2Container = memo(() => <div>Sample 2 Component.</div>);

export { Sample2Container };
