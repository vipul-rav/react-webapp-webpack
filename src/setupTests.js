import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

let store;
beforeEach(() => {
  const mockStore = configureMockStore();
  store = mockStore();
});

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
