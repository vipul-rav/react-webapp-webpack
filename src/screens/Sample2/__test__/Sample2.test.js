import React from 'react';
import { cleanup } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { render } from '../../../setupTests';
import { Sample2Container } from '../Sample2Container';

const content = {
  ExitText: 'Exit',
  btnContinue: 'Continue',
  homePageText: 'Home Page'
};

const setup = (overridesProps) => {
  const props = {
    content: content,

    ...overridesProps
  };

  const mockStore = configureMockStore();
  const store = mockStore({
    config: {
      externalContent: content
    }
  });

  const { container, getByText } = render(<Sample2Container {...props} store={store} />);

  return {
    props,
    container,
    getByText
  };
};

describe('Sample2 container', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
