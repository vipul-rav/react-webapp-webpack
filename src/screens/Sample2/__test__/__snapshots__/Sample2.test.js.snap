// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample2 container should render 1`] = `
<div>
  <div>
    Sample 2 Component.
  </div>
</div>
`;
